### PR TITLE
[#1628] test(client): Fix double release ShuffleIndexResult cause exception to pass flaky test 

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes were proposed in this pull request?

(Please outline the changes and how this PR fixes the issue.)

### Why are the changes needed?

- Pass the flaky test
- Avoid memory leak after double release the `ShuffleIndexResult`

Fix: #1628

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Flaky test can be fixed.

### CODE REVIEW

File: common/src/main/java/org/apache/uniffle/common/ShuffleIndexResult.java [hunk 1]
```diff
@@ -74,7 +78,15 @@ public boolean isEmpty() {
 
   public void release() {
     if (this.buffer != null) {
-      this.buffer.release();
+      try {
+        this.buffer.release();
+      } catch (Throwable e) {
+        LOG.warn(
```
Comment (jerqi) [inline common/src/main/java/org/apache/uniffle/common/ShuffleIndexResult.java:84]: Could you log the `Throwable e`, too?

Comment (maobaolong) [reply common/src/main/java/org/apache/uniffle/common/ShuffleIndexResult.java:84]: Sure

**Comment by maobaolong** [comment]
@jerqi Log the e too, PTAL.

File: common/src/main/java/org/apache/uniffle/common/ShuffleIndexResult.java (hunk 1 above)
Comment (jerqi) [reply common/src/main/java/org/apache/uniffle/common/ShuffleIndexResult.java:84]: You should catch specific exception class instead of Throwable.

**Review by jerqi — APPROVED** [review]
LGTM.

**Comment by maobaolong** [comment]
@jerqi Thanks for your review!

**Pull request merged**